src/common/internet.rs

```diff
@@ -21,6 +21,13 @@ use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 use torut::onion::{TorPublicKeyV3, OnionAddressV3, TORV3_PUBLIC_KEY_LENGTH};
 
 
+pub trait UniformAddrEncodable<const L: usize>: Sized {
```

Comment:
As a casual bypasser that is just looking, I can tell you that this could really use a comment explaining what it is for and why does it exist. :)

Reply:
It does not exist already :) Rust compiler is still very buggy with const generics